akka-actor/src/main/scala/akka/util/OptionVal.scala

```diff
@@ -9,13 +9,15 @@ package akka.util
  */
 private[akka] object OptionVal {
 
-  def apply[A >: Null](x: A): OptionVal[A] = new OptionVal(x)
+  def apply[A](x: A): OptionVal[A] = new OptionVal(x)
 
   object Some {
-    def apply[A >: Null](x: A): OptionVal[A] = new OptionVal(x)
-    def unapply[A >: Null](x: OptionVal[A]): OptionVal[A] = x
+    def apply[A](x: A): OptionVal[A] = new OptionVal(x)
+    def unapply[A](x: OptionVal[A]): OptionVal[A] = x
   }
 
+  def none[A]: OptionVal[A] = None.asInstanceOf[OptionVal[A]]
+
```

Comment:
That looks too simple, but fantastic if it works.
The purpose is to be allocation free for both creating `Some(a)` and for
```
opt match {
  case OptionVal.Some(a) =>
  case OptionVal.None =>
}
```

Reply:
Yep, it should all work as before.

One thing that's different (I mean, even before my change) is there's no match exhaustion, so if nothing saves you from forgetting the `OptionVal.Some` or `OptionVal.None` case.  And sometimes you can have duplicates, like multiple `case OptionVal.None` 😄

Reply:
Moved to #25868